plugins/aea-ledger-ethereum/tests/test_ethereum_contract.py

```diff
@@ -66,3 +66,47 @@ def test_get_contract_instance(ethereum_testnet_config, ganache):
     assert (
         str(type(instance)) == "<class 'web3._utils.datatypes.PropertyCheckingFactory'>"
     )
+
+
+@pytest.mark.flaky(reruns=MAX_FLAKY_RERUNS)
+@pytest.mark.integration
+@pytest.mark.ledger
+def test_gas_station_strategy(ethereum_testnet_config, ganache):
+    """Test the get contract instance method."""
+    ec = EthereumCrypto(private_key_path=ETHEREUM_PRIVATE_KEY_PATH)
+
+    ethereum_api = EthereumApi(**ethereum_testnet_config)
+    full_path = Path(ROOT_DIR, "tests", "data", "dummy_contract", "build", "some.json")
+    contract_interface = ethereum_api.load_contract_interface(full_path)
+    tx = ethereum_api.get_deploy_transaction(
+        contract_interface,
+        ec.address,
+        0,
+        gas_price_strategy="gas_station"
+    )
+    assert all([
+        key in tx
+        for key in ['gas', 'chainId', 'value', 'nonce', 'gasPrice', 'data', 'from']
+    ])
+
+
+@pytest.mark.flaky(reruns=MAX_FLAKY_RERUNS)
+@pytest.mark.integration
+@pytest.mark.ledger
+def test_eip1559_strategy(ethereum_testnet_config, ganache):
+    """Test the get contract instance method."""
+    ec = EthereumCrypto(private_key_path=ETHEREUM_PRIVATE_KEY_PATH)
+
+    ethereum_api = EthereumApi(**ethereum_testnet_config)
+    full_path = Path(ROOT_DIR, "tests", "data", "dummy_contract", "build", "some.json")
+    contract_interface = ethereum_api.load_contract_interface(full_path)
+    tx = ethereum_api.get_deploy_transaction(
+        contract_interface,
+        ec.address,
+        0,
+        gas_price_strategy="eip1559"
+    )
+    assert all([
+        key in tx
+        for key in ['gas', 'gasPrice', 'chainId', 'value', 'nonce', 'maxFeePerGas', 'maxPriorityFeePerGas', 'baseFee', 'data', 'from']
```

Comment:
It should never be the case that you have `gasPrice` and `maxFeePerGas`/`maxPriorityFeePerGas`. So here we should only have the latter two